modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,46 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
-
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
-					continue;
-				}
-
 				var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
 				for ( var j = 0; j < images.length; j++ ) {
-					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
+
+					if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
+						continue;
+					}
+
+					var sizes = media[i].media_details.sizes;
+					var sizes_keys = Object.keys( sizes );
 					var srcset = images[j].getAttribute( 'srcset' );
-					if ( srcset ) {
-						images[j].setAttribute( 'srcset', srcset.replace( /\.webp(\s)/ig, ext[0] + '$1' ) );
+
+					// If a full image is present in srcset, it should be updated.
+					if( srcset && media[i].media_details.sources['image/webp'] ) {
+						srcset = srcset.replace( media[i].media_details.sources['image/webp'].file, media[i].media_details.sources['image/jpeg'].file );
+					}
+
+					var flag = true;
+					for ( var k = 0; k < sizes_keys.length; k++ ) {
```

Comment:
if there is no srcset, we can just look at the full size image and skip this loop

Reply:
Thank you for the review.

- If there is no `srcset` then the script will skip the search we check [the condition](https://github.com/WordPress/performance/blob/fix/426-enhance-js-replacement-mechanism/modules/images/webp-uploads/fallback.js#L38-L40) for all srcset data. 

- The `flag` is set to `false` once the img src is updated and it will skip the checking again for other sizes.

Reply:
+1 to what @adamsilverstein said. Trying to clarify (the way I understand it):
* First of all, we should probably store `src` in a variable, similar to how we store `srcset` in a variable, before the `sizes_keys` loop.
* In the if clause where we adjust the `src` attribute, we can then check if there is no `srcset` attribute, and if so we can completely `break` out of the loop. Because if there is only `src` and no `srcset` and we have already replaced the `src`, we know for sure that there is no more image URL to replace.

Let me know if that makes more sense.

Reply:
👍🏼 thanks for clarifying that is exactly what I meant @felixarntz

Reply:
Thank you for explaining 😄 Adam I get your point now.

This feedback is addressed.